spectator/meter/counter.go

```diff
@@ -8,54 +13,39 @@ package spectator
 //
 // https://netflix.github.io/spectator/en/latest/intro/counter/
 type Counter struct {
-	count uint64
-	// Pointers need to be after counters to ensure 64-bit alignment. See
-	// note in atomicnum.go
-	id    *Id
+	id              *Id
+	writer          writer.Writer
+	meterTypeSymbol rune
 }
 
 // NewCounter generates a new counter, using the provided meter identifier.
-func NewCounter(id *Id) *Counter {
-	return &Counter{0, id}
+func NewCounter(id *Id, writer writer.Writer) *Counter {
+	return &Counter{id, writer, 'c'}
 }
 
 // MeterId returns the meter identifier.
 func (c *Counter) MeterId() *Id {
 	return c.id
 }
 
-// Measure returns the list of measurements known by the counter. This will
-// either contain one item, the last value, or no items. This also resets the
-// internal counter to 0.0.
-func (c *Counter) Measure() []Measurement {
-	cnt := swapFloat64(&c.count, 0.0)
-	if cnt > 0 {
-		return []Measurement{{c.id.WithDefaultStat("count"), cnt}}
-	} else {
-		return []Measurement{}
-	}
-}
-
 // Increment increments the counter.
 func (c *Counter) Increment() {
-	addFloat64(&c.count, 1)
+	var line = fmt.Sprintf("%c:%s:%d", c.meterTypeSymbol, c.id.spectatordId, 1)
+	c.writer.Write(line)
 }
 
 // AddFloat adds a specific float64 delta to the current measurement.
 func (c *Counter) AddFloat(delta float64) {
```

Comment:
Given that function overloading does not exist in Go, this is probably the most straight-forward thing to do, and matches the behavior of the previous implementation.

I was wondering if we could use the new variadic functions feature with an interface to simplify this, but given that this was part of the original API, that would likely break things.